src/airtable/project_summary.js

```diff
@@ -28,12 +34,23 @@ const deleteAll = async () => {
 }
 
 const processAll = async () => {
-  const proposals = await retrieve.proposals()
-  const projects = summarize(proposals)
-  const entries = toAirtableList(projects)
-  const chunks = chunk(entries)
-  for (const c of chunks) {
-    await populate(c)
+  try {
+    let proposals = await retrieve.proposals()
+
+    // Updates all proposals that don't have UUID (it adds one)
+    await addUUID(proposals)
+
+    // Retrieve all proposal (now all with UUID)
+    proposals = await retrieve.proposals()
```

Comment:
Could return a value from `await addUUID(proposals)` instead of calling `await retrieve.proposals()` twice.

Reply:
Hmm, I'm not sure it will work. We retrieve the proposals a second time since we've updated some or all records (one by one) within the main (`Proposals`) table through the `addUUID` function. It complicates the code less if we just fetch the proposals again. Maybe if we add a check for newly created UUIDs we can skip the 2nd fetch when it’s not needed.

Reply:
Then let's add a check to skip if the UUIDs exist.

Reply:
Done